journalapp/scripts/load_api.py

```diff
@@ -0,0 +1,109 @@
+# coding=utf-8
+import os
+import requests
+
+from journalapp.models import Entry
+from sqlalchemy import engine_from_config
+from sqlalchemy.orm import scoped_session, sessionmaker
+from zope.sqlalchemy import ZopeTransactionExtension
+
+
+DBSession = scoped_session(sessionmaker(extension=ZopeTransactionExtension()))
+
+
+def load_with_key():
+    """
+    Opens a DB session
+    Collects and processes a JSON into a DB session
+    Closes the session
+    """
+    # Get API key
+    api_key = os.environ.get('API_KEY', None)
+
+    # Get DB Session
+    database_url = os.environ.get('DATABASE_URL', None)
+    print(database_url)
+    engine = engine_from_config({'sqlalchemy.url': database_url})
+    DBSession.configure(bind=engine)
+
+    json_payload = {}
+    r = requests.get('https://sea401d2.crisewing.com/api/export?apikey=' + api_key)
+    if r.status_code == requests.codes.ok:
+        json_payload = r.json()
+    if json_payload:
+        """post to journal"""
+        for entry in json_payload:
+            load_entry(entry["title"], entry["text"], entry["created"])
+
+    # Flush DB Session
+    print("Before:\n", DBSession.new)
+    DBSession.flush()
+    print("After:\n", DBSession.new)
+
+
+def load_entry(update_title, update_text, update_created=None):
+    """
+    ADDs or UPDATEs an entry in the DB session
+    """
+    try:
+        """ Try to find an item in the DB, if found update """
+        existing_entry = DBSession.query(Entry).filter(Entry.title == update_title)
+        update_dict = {
+            "title": update_title,
+            "text": update_text,
+            "created": update_created
+        }
+        DBSession.query(Entry).filter(Entry.id == existing_entry.id).update(update_dict)
+        print("Updated item:\n{}\n{}\n{}\n\n".format(**update_dict))
```

Comment:
Interesting choice here.  Allowing you to update body text on entries that have the same title?